API.md

```diff
@@ -2366,8 +2366,65 @@ Defines a route rules processor for converting route rules object into route con
 Note that the root server and each plugin server instance can only register one rules processor.
 If a route is added after the rules are configured, it will not include the rules config. Routes
 added by plugins apply the rules to each of the parent realms' rules from the root to the route's
-realm. This means the processor defined by the plugin override the config generated by the root
-processor if they overlap. The route `config` overrides the rules config if the overlap.
+realm. This means the processor defined by the plugin overrides the config generated by the root
+processor if they overlap. `route.config` overrides `route.rules` if the resulting processed config overlaps.
+
+```js
+const validateSchema = {
+  auth: Joi.string(),
+  myCustomPre: Joi.array().min(2).items(Joi.string()),
+  payload: Joi.object()
+}
+
+const processor = (rules, info) => {
+
+    if (!rules) {
+        return null;
+    }
+
+    const options = {};
+
+    if (rules.auth) {
+        options.auth = {
+            strategy: rules.auth,
+            validate: {
+                entity: 'user'
+            }
+        }
+    }
+
+    if (rules.myCustomPre) {
+
+        options.pre = [
+            myPreHelper(...rules.myCustomPre)
+        ];
+    }
+
+    if (rules.payload) {
+        options.validate.payload = Joi.object(rules.payload);
+    }
+
+    return options;
+};
+
+const realmRules = {
+    validate: { schema: validateSchema }
+};
+
+server.rules(processor, realmRules);
```

Comment:
The goal here is just to make things more concise and avoid the "realms" reference for those unfamiliar with that hapi concept.

```suggestion
server.rules(processor, {
    validate: { schema: validateSchema }
});
```

Reply:
I will update this when I get a chance some time this week. Thank you for the feedback!